src/main/java/ca/uhn/fhir/jpa/starter/AuthenticationInterceptor.java

```diff
@@ -0,0 +1,56 @@
+package ca.uhn.fhir.jpa.starter;
+
+import java.nio.charset.StandardCharsets;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+
+import ca.uhn.fhir.interceptor.api.Hook;
+import ca.uhn.fhir.interceptor.api.Interceptor;
+import ca.uhn.fhir.interceptor.api.Pointcut;
+import ca.uhn.fhir.rest.server.exceptions.AuthenticationException;
+
+@Interceptor
+public class AuthenticationInterceptor {
+
+
+	@Hook(Pointcut.SERVER_INCOMING_REQUEST_POST_PROCESSED)
+	public boolean incomingRequestPostProcessed(HttpServletRequest theRequest,
+			HttpServletResponse theResponse) throws AuthenticationException {
+
+		String authHeader = theRequest.getHeader("Authorization");
+
+		// The format of the header must be:
+		// Authorization: Basic [base64 of username:password]
+		if (authHeader == null) {
+			throw new AuthenticationException("Missing Authorization header");
+		}
+		if (!authHeader.startsWith("Basic")) {
+			throw new AuthenticationException("Invalid Authorization header.Missing Basic prefix");
+		}
+
+		if (!StringUtils.isBlank(authHeader) && authHeader.startsWith("Basic")) {
+			String base64 = authHeader.substring(6);
+			String base64decoded = new String(Base64.decodeBase64(base64), StandardCharsets.UTF_8);
+			String[] parts = base64decoded.split(":");
+
+			String username = parts[0].toLowerCase();
+			String password = parts[1];
+
+			/*
+			 * Here we test for a hardcoded username & password use to get access to the
+			 * hapi fhir JPA sever..
+			 */
+			if (!username.equals("hapi") & !password.equals("hapi123")) {
```

Comment:
@gitcliff   here you used a bitwsie operator instead of the Logical operator

Reply:
Also, this should be `||` (logical or) not `&&` (logical and)